internal/libs/autofile/autofile.go

```diff
@@ -43,8 +44,12 @@ const (
 //
 // This is useful for using a log file with the logrotate tool.
 type AutoFile struct {
-	ID   string
-	Path string
+	// Putting this file nWrites at the beginning so that it is
+	// ALWAYS byte aligned to be safely used in atomics, given
+	// that we need it super fast and able to check
+	nWrites uint64
```

Comment:
this is really just used as a bool (zero or not zero?) 

Seems likely that the mutex that protects this, let alone any of the actual IO that we're doing are enough orders of magnitude slower that this atomic/alignment business is probably not worth the sacrifice in clarity, but I'm not sure.

Reply:
>this is really just used as a bool (zero or not zero?)

Yup, but there isn't an atomic boolean and in the case of fast writes can count the writes as well.

>Seems likely that the mutex that protects this, let alone any of the actual IO that we're doing are enough orders of magnitude slower that this atomic/alignment business is probably not worth the sacrifice in clarity, but I'm not sure.

Alignment doesn't have anything to do with the mutex; we can get panics if we try to use atomics with non-aligned fields regardless of if we have a mutex or not and it is documented at https://pkg.go.dev/sync/atomic#pkg-note-BUG
![Screen Shot 2021-12-08 at 4 46 39 PM](https://user-images.githubusercontent.com/4898263/145314184-cc75a498-b7f7-4a94-b44d-fd72802b4a25.png)

>magnitude slower that this atomic/alignment business is probably not worth the sacrifice in clarity, but I'm not sure.

So what decision would you prefer then?

Reply:
I'd just use a boolean for clarity given that operations on this value are always protected by a mutex.

Reply:
Alright, thanks @tychoish, I've added a boolean. Please take a look.